File: jscomp/test/nested_obj_test.js
'use strict';


var f = do
  x: do
    y: do
      z: 3
    end
  end
end;

var f2_000 = --[ :: ]--[
  do
    x: do
      y: do
        z: 3
      end
    end
  end,
  --[ :: ]--[
    do
      x: do
        y: do
          z: 31
        end
      end
    end,
    --[ [] ]--0
  ]
];

var f2_001 = [
  do
    x: do
      y: do
        z: 3
      end
    end
  end,
  do
    x: do
      y: do
        z: 31
      end
    end
  end
];

var f2 = --[ tuple ]--[
  f2_000,
  f2_001
];

var f3 = do
  x: do
    y: do
      z: 3
    end
  end
end;

var f_record = do
  x: do
    y: do
      z: 3
    end
  end
end;

exports.f_record = f_record;
exports.f = f;
exports.f2 = f2;
exports.f3 = f3;
--[ No side effect ]--

File: jscomp/test/nested_obj_test.lua
__console = {log = print};


f = {
  x = {
    y = {
      z = 3
    }
  }
};

f2_000 = --[[ :: ]]{
  {
    x = {
      y = {
        z = 3
      }
    }
  },
  --[[ :: ]]{
    {
      x = {
        y = {
          z = 31
        }
      }
    },
    --[[ [] ]]0
  }
};

f2_001 = {
  {
    x = {
      y = {
        z = 3
      }
    }
  },
  {
    x = {
      y = {
        z = 31
      }
    }
  }
};

f2 = --[[ tuple ]]{
  f2_000,
  f2_001
};

f3 = {
  x = {
    y = {
      z = 3
    }
  }
};

f_record = {
  x = {
    y = {
      z = 3
    }
  }
};

exports = {};
exports.f_record = f_record;
exports.f = f;
exports.f2 = f2;
exports.f3 = f3;
return exports;
--[[ No side effect ]]
